### PR TITLE
Spring Boot 2.1.2 + JUnit 5

### PR DESCRIPTION
**Pull request by maximAtanasov**
Contains PR #183 as well.

### CODE REVIEW

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metric/MetricRegistryBean.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.wickedsource.coderadar.metric;
+
+import com.codahale.metrics.MetricRegistry;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MetricRegistryBean extends MetricRegistry {}
```
Comment (thombergs) [inline coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metric/MetricRegistryBean.java]: what do we need this bean for?

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java [hunk 2]
```diff
@@ -51,6 +53,12 @@ protected void configure(AuthenticationManagerBuilder auth) throws Exception {
     auth.userDetailsService(userDetailsService).passwordEncoder(getPasswordEncoder());
   }
 
+  @Bean(name = BeanIds.AUTHENTICATION_MANAGER)
```
Comment (thombergs) [inline coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java]: why did you introduce this bean?

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/user/service/RegistrationService.java [hunk 3]
```diff
@@ -54,10 +55,10 @@ public boolean userExists(UserRegistrationDataResource userRegistrationDataResou
    * @return User with id <code>userId</code>
    */
   public User getUser(Long userId) {
-    User user = userRepository.findOne(userId);
-    if (user == null) {
+    Optional<User> user = userRepository.findById(userId);
+    if (!user.isPresent()) {
       throw new ResourceNotFoundException();
```
Comment (thombergs) [inline coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/user/service/RegistrationService.java]: again, `ResourceNotFoundException` introduces a dependency to the rest layer. 
Perhaps move `ResourceNotFoundException` to the package `core.common` in order to avoid this dependency.

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/analyzingjob/rest/AnalyzingJobControllerTest.java [hunk 4]
```diff
@@ -14,13 +14,12 @@
...
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 import org.springframework.http.MediaType;
-import org.wickedsource.coderadar.testframework.category.ControllerTest;
 import org.wickedsource.coderadar.testframework.template.ControllerTestTemplate;
 
-@Category(ControllerTest.class)
+@Tag("ControllerTest.class")
```
Comment (thombergs) [inline coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/analyzingjob/rest/AnalyzingJobControllerTest.java]: "ControllerTest.class" is a weird tag. 
Perhaps add a public constant `TAG` to the `ControllerTestTemplate` class containing the String `ControllerTest` and use this within the `@Tag` annotation.

Also: can't we just add the tag to the parent class `ControllerTestTemplate` and remove it from the child classes?

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java [hunk 5]
```diff
@@ -79,22 +74,23 @@ public void cleanup() {
   }
 
   @Test
-  @Category(IntegrationTest.class)
+  @Tag("IntegrationTest.class")
```
Comment (thombergs) [inline coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java]: same as with "ControllerTest.class"

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeControllerTest.java [hunk 6]
```diff
@@ -61,11 +61,11 @@ public void queryMetricsTree() throws Exception {
                             .type(JsonFieldType.STRING)
                             .description(
                                 "Either 'MODULE' if this node describes a module which can have child nodes or 'FILE' if this node describes a file (which has no child nodes)."),
-                        fieldWithPath("metrics")
+                        subsectionWithPath("metrics")
```
Comment (thombergs) [inline coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeControllerTest.java]: why did you change `fieldWithPath` to `subsectionWithPath`?

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/core/WorkdirManager.java [hunk 7]
```diff
@@ -29,10 +31,14 @@ public WorkdirManager(CoderadarConfiguration config, ProjectRepository projectRe
...
-    Path workdir = config.getWorkdir().resolve("projects/" + project.getWorkdirName());
-    createDirIfNecessary(workdir);
-    return workdir;
+    Optional<Project> project = projectRepository.findById(projectId);
+    if (project.isPresent()) {
+      Path workdir = config.getWorkdir().resolve("projects/" + project.get().getWorkdirName());
+      createDirIfNecessary(workdir);
+      return workdir;
+    } else {
+      throw new ResourceNotFoundException("Project with id:" + projectId + " not found");
```
[the remark this answers is not in the document]
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/core/WorkdirManager.java]: I have moved the Exception to the `core.common` package. Should we still throw it or use a `RuntimeException` instead?

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/core/monitoring/PrometheusConfiguration.java [hunk 8]
```diff
@@ -10,26 +11,29 @@
...
 import org.springframework.context.annotation.Configuration;
 
+*/
 /**
  * This configuration activates the "/prometheus" endpoint and exposes default metrics and all
  * DropWizard metrics to this endpoint.
- */
-@Configuration
-@EnablePrometheusEndpoint
-public class PrometheusConfiguration {
```
[the remark this answers is not in the document]
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/core/monitoring/PrometheusConfiguration.java]: Yes, commenting out the class was a mistake. It got rid of some errors but it seems I forgot to 
fix Prometheus afterwards.

It seems that the old Prometheus Configuration is no longer supported in Spring-Boot 2 as explained in this post: https://github.com/prometheus/client_java/issues/345

I have applied the fixes from this SO post https://stackoverflow.com/questions/48700449/cannot-include-prometheus-metrics-in-spring-boot-2-version-2-0-0-m7/48766683#48766683
and started a local Prometheus instance. Metrics seem to be available at `/actuator/prometheus` and Prometheus properly recognizes them. I'm not familiar with how to use the Prometheus server to browse metrics I cannot verify that everything is 100% correct, but it does seem to work.

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metric/MetricRegistryBean.java (hunk 1 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metric/MetricRegistryBean.java]: Coderadar requires a MetricRegistry bean to be available, but it is no longer automatically created, therefore I created it manually.

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java (hunk 2 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java]: An AuthenticationManager Bean is required, but is no longer automatically created, so I created it myself. 
https://github.com/spring-projects/spring-boot/issues/11136 According to this conversation, the bean must be created manually when a custom `UserDetailsService` is present

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/user/service/RegistrationService.java (hunk 3 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/user/service/RegistrationService.java]: I have moved the Exception to the `core.common` package.

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/analyzingjob/rest/AnalyzingJobControllerTest.java (hunk 4 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/analyzingjob/rest/AnalyzingJobControllerTest.java]: I have done this

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java (hunk 5 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java]: Done

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeControllerTest.java (hunk 6 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeControllerTest.java]: Otherwise I get the following error:

```
org.springframework.restdocs.snippet.SnippetException: The following parts of the payload were not documented:
{
  "metrics" : {
    "metric1" : 38,
    "metric2" : 28,
    "metric3" : 19
  }
}
```

I suspect that is because we don't document `metric1`, `metric2` and `metric3` in `requestFields`.
Seeing as how the test passes all other checks (including cheking the values of the metrics), I thought this change was appropriate.

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/core/monitoring/PrometheusConfiguration.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (thombergs) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/core/monitoring/PrometheusConfiguration.java]: looks good :)

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java (hunk 5 above)
Comment (thombergs) [reply coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java]: @maximAtanasov I see no change

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeControllerTest.java (hunk 6 above)
Comment (thombergs) [reply coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeControllerTest.java]: OK

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java (hunk 2 above)
Comment (thombergs) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/security/CoderadarSecurityConfiguration.java]: OK

File: coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metric/MetricRegistryBean.java (hunk 1 above)
Comment (thombergs) [reply coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metric/MetricRegistryBean.java]: OK

**Comment by thombergs** [comment]
@maximAtanasov  one last change (see open comment)

File: coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java (hunk 5 above)
Comment (maximAtanasov) [reply coderadar-server/coderadar-core/src/test/java/org/wickedsource/coderadar/job/scan/commit/CommitMetadataScannerIntegrationTest.java]: Sorry, I missed that tag. I've changed it to simply `IntegrationTest` now.
